org.hl7.fhir.dstu2/src/main/java/org/hl7/fhir/dstu2/model/BaseDateTimeType.java

```diff
@@ -618,4 +620,9 @@ protected void setValueAsV3String(String theV3String) {
 		}
 	}
 
+  private TimeZone getTimeZone(String offset) {
+    return timezoneCache.computeIfAbsent(offset, (offsetLocal) ->
+      TimeZone.getTimeZone("GMT" + offsetLocal));
+  }
```

Comment:
It took me far too long for me to figure out why this wasn't working. You're passing in "GMTGMT-offset" here as the function parameter instead of just "GMT-offset", so the BaseDateTime is just detecting the GMT and not taking into account the offset when storing the timezone. This results in the timezone based fhirpath tests failing. You should replace it with this:

```
  private TimeZone getTimeZone(String offset) {
    return timezoneCache.computeIfAbsent(offset, TimeZone::getTimeZone);
  }
```

With that change, it will pass all tests.